commons-data-impl/src/main/java/com/appjars/saturn/dao/jpa/ConversionJpaDaoSupport.java

```diff
@@ -22,7 +22,6 @@
 import java.io.Serializable;
```

Comment:
Why this one was not removed?

Reply:
Section 2.4 of JSR-317 (Primary Keys and Entity Identity) specifies that 

> The following rules apply for composite primary keys:
> (...) The primary key class must be serializable.

Additionally:
> A simple primary key or a field or property of a composite primary key should be one of the following types: any Java primitive type; any primitive wrapper type; java.lang.String; java.util.Date; java.sql.Date; java.math.BigDecimal; java.math.BigInteger. 

Therefore, the fact that the key must be serializable is a constraint introduced by JPA (but it doesn't carry on to the presentation/service layers nor other persistence implementations). Implementations of `ConversionJpaDaoSupport` (or `JpaDaoSupport`) will have a key with an upper bound of `Serializable`, while the upper bound will be `Object` elsewhere (unless another non-JPA constraint apply)

Reply:
Excellent, solved then